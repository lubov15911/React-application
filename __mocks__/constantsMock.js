export const moviePreviewData = {
    poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
    title: 'Coco',
    vote_average: 7.8,
    tagline: 'The celebration of a lifetime',
    release_date: '2017-10-27',
    runtime: 105,
    genres: ["Adventure", "Comedy", "Family", "Animation"],
    overview: 'Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz.',
};

export const filmsData = [{
    id: 354912,
    poster_path: 'https://image.tmdb.org/t/p/w500/eKi8dIrr8voobbaGzDpe8w0PVbC.jpg',
    title: 'Coco',
    release_date: '2017-10-27',
    genres: ["Adventure", "Comedy", "Family", "Animation"],
}, {
    id: 269149,
    poster_path: "https://image.tmdb.org/t/p/w500/sM33SANp9z6rXW8Itn7NnG1GOEs.jpg",
    title: "Zootopia",
    release_date: "2016-02-11",
    genres: ["Animation", "Adventure", "Family", "Comedy"],
}, {
    id: 333339,
    poster_path: 'https://image.tmdb.org/t/p/w500/pU1ULUq8D3iRxl1fdX2lZIzdHuI.jpg',
    title: 'Ready Player One',
    release_date: '2018-03-28',
    genres: ["Adventure", "Science Fiction", "Action"],
}, {
    id: 337167,
    poster_path: "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
    title: "Fifty Shades Freed",
    release_date: "2018-03-28",
    genres: ["Drama", "Romance"],
}];

export const searchOptionsData = {
    title: 'Title',
    genre: 'Genre'
};

export const sortOptionsData = {
    rating: 'Rating',
    release: 'Release date'
};

export const searchValueData = 'Ready';
