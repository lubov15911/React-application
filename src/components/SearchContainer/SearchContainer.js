import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import ToggleComponent from '../ToggleComponent';
import SearchBar from '../SearchBar';

import './SearchContainer.scss';

import { SearchOptions } from '../../constants';
import { updateSearchOption } from '../../actions';

const propTypes = {
    searchOption: PropTypes.string.isRequired,
    toggleSearchOption: PropTypes.func.isRequired,
};

const SearchContainer = ({ searchOption, toggleSearchOption, }) => {
    const handleToggle = ({ currentTarget: { value } }) => {
        toggleSearchOption(value);
    };

    return (
        <div className="search-header">
            <h1 className="search-title">Find your movie</h1>
            <SearchBar />
            <ToggleComponent
                toggleType="Search "
                options={SearchOptions}
                selected={searchOption}
                handleToggle={handleToggle}/>
        </div>
    )
};
SearchContainer.propTypes = propTypes;

const mapStateToProps = (state) => {
    return {
        searchOption: state.searchOption,
    }
};

export default connect(mapStateToProps, { toggleSearchOption: updateSearchOption })(SearchContainer);
