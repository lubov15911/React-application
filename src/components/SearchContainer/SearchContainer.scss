.search-header {
  display: flex;
  flex-direction: column;
  margin: 0 6em;

  .search-title {
    font-size: 4em;
    font-weight: normal;
  }
}
