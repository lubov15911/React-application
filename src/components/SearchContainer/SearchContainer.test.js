import React from 'react';
import { shallow } from 'enzyme';

import SearchContainer from './SearchContainer';

import { searchValueData, searchOptionsData } from '../../../__mocks__/constantsMock';

describe('SearchContainer', () => {
    const simpleFakeFunction = () => {};
    const spyFakeFunctionSearchSubmit = jest.fn();

    let component;

    beforeEach(() => {
        component = shallow(<SearchContainer
            handleSearchValue={simpleFakeFunction}
            handleSearchSubmit={spyFakeFunctionSearchSubmit}
            toggleSearch={simpleFakeFunction}
            searchOption={searchOptionsData.title}
            searchValue={searchValueData}
        />);
    });

    it('should render correctly with props', () => {
        expect(component).toMatchSnapshot();
    });

    it('should change state of error - true', () => {
        let error;
        try {
            component.find('SearchBar').props().handleSubmit('error');
            expect(component.state().error).toBeTruthy();
        } catch (e) {
            error = e;
        }
        expect(error).toBeInstanceOf(Error);
    });

    it('should change state of error - false', () => {
        let error;
        try {
            component.find('SearchBar').props().handleSubmit(searchValueData);
            expect(component.state().error).toBeFalsy();
        } catch (e) {
            error = e;
        }
        expect(error).not.toBeInstanceOf(Error);
    });

    it('should call search request', () => {
        component.find('SearchBar').props().handleSubmit(searchValueData);
        expect(spyFakeFunctionSearchSubmit).toHaveBeenCalledWith(searchValueData.toLocaleLowerCase());
    });
});
