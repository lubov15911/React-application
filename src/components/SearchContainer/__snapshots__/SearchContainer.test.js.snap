// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer should render correctly with props 1`] = `
<div
  className="search-header"
>
  <h1
    className="search-title"
  >
    Find your movie
  </h1>
  <Connect(SearchBar) />
  <ToggleComponent
    handleToggle={[Function]}
    options={
      Object {
        "first": "Title",
        "second": "Genres",
      }
    }
    selected="Title"
    toggleType="Search "
  />
</div>
`;
